karabo_data/geometry2.py

```diff
@@ -631,6 +695,229 @@ def plot_data(self, modules_data):
 {name}/coffset = {coffset}
 """
 
+class LPD_1MGeometry(DetectorGeometryBase):
+    """Detector layout for LPD-1M
+
+    The coordinates used in this class are 3D (x, y, z), and represent multiples
+    of the pixel size.
+    """
+    pixel_size = 5e-4  # 5e-4 metres == 0.5 mm
+    frag_ss_pixels = 32
+    frag_fs_pixels = 128
+    expected_data_shape = (16, 256, 256)
+
+    @classmethod
+    def from_quad_positions(cls, quad_pos, *, unit=1e-3, asic_gap=None,
+                            panel_gap=None):
+        """Generate an LPD-1M geometry from quadrant positions.
+
+        This produces an idealised geometry, assuming all modules are perfectly
+        flat, aligned and equally spaced within their quadrant.
+
+        The quadrant positions refer to the corner of each quadrant
+        where module 4, tile 16 is positioned.
+        This is the corner of the last pixel as the data is stored.
+        In the initial detector layout, the corner positions are for the top
+        right corner of the quadrant, looking into the beam.
+
+        Parameters
+        ----------
+        quad_pos: list of 2-tuples
+          (x, y) coordinates of the last corner (the one by module 4) of each
```

Comment:
I want to identify which corner of the quadrant you're giving the position for without referring to left and right, because they depend on which way you plot the image. But in [the diagram here](https://in.xfel.eu/readthedocs/docs/detector-documentation/en/latest/lpd/source/introduction.html), for instance, they're the top left corners of each quadrant.

Each LPD quadrant has 4 (super)modules, numbered 1-4. Whichever way you plot the image, each quadrant is positioned by its corner nearest to module 4 in that quadrant.

I'm scribbling lots of little diagrams on pieces of paper as I work on this to make sense of what's where. It's really hard to describe clearly. I'm starting to think we should borrow the terms 'port' and 'starboard', because sailors solved exactly the same problem with left/right centuries ago.